packages/dd-trace/src/opentracing/propagation/text_map.js

```diff
@@ -285,14 +285,17 @@ class TextMapPropagator {
         case 'tracecontext':
           spanContext = this._extractTraceparentContext(carrier)
           break
-        case 'b3' && this
-          ._config
-          .tracePropagationStyle
-          .otelPropagators: // TODO: should match "b3 single header" in next major
         case 'b3 single header': // TODO: delete in major after singular "b3"
           spanContext = this._extractB3SingleContext(carrier)
           break
         case 'b3':
+          if (this._config.tracePropagationStyle.otelPropagators) {
+            // TODO: should match "b3 single header" in next major
+            spanContext = this._extractB3SingleContext(carrier)
+          } else {
+            spanContext = this._extractB3MultiContext(carrier)
+          }
+          break
```

Comment:
@khanayan123 You added the original lines here, so can you verify that this logic is correct?

Reply:
This looks right to me